src/store/reducers/appReducer.ts

```diff
@@ -0,0 +1,60 @@
+import { APP_TYPES } from "../types";
+import { Movie, ReduxAction } from "../../components/MovieCard/interface";
+
+const initialState = {
+    isAddModalOpened: false,
+    isEditModalOpened: false,
+    isDeleteModalOpened: false,
+    viewMode: "search",
+    detailedViewData: {},
+}
+
+export default function appReducer(state: typeof initialState = initialState, action: ReduxAction) {
+    switch (action.type) {
+        case APP_TYPES.TOGGLE_MODAL: {
+            if (action.payload.status) {
+                document.documentElement.classList.add("no-scroll")
```

Comment:
You shouldn't be throwing any side-effects from within your reducer as it has to stay pure function. You may probably want to perform these inside appropriate middleware.